packages/constellation/src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -1,26 +1,24 @@
-/* eslint-disable import/no-unresolved */
 import '../styles.css'
 
-import React, { CSSProperties, useMemo } from 'react'
+import React, { useEffect } from 'react'
 
+import { DarkModeContext } from './themeContext'
 import { ThemeProviderComponent } from './ThemeProvider.types'
 
-const ThemeProvider: ThemeProviderComponent = ({ children, theme, colorOverride = {} }) => {
-  const colorOverrideStyles = useMemo((): CSSProperties => {
-    return Object.keys(colorOverride).reduce(
-      (newColorOverride, color) => ({
-        ...newColorOverride,
-        [`--color-${color}`]: colorOverride[color as keyof typeof colorOverride],
-      }),
-      {},
+const ThemeProvider: ThemeProviderComponent = ({ children, darkMode = false, theme = {} }) => {
+  useEffect(() => {
+    document.body.classList.add(darkMode ? 'mode-dark' : 'mode-light')
+    document.body.classList.remove(darkMode ? 'mode-light' : 'mode-dark')
+
+    const themeStyles = Object.keys(theme).reduce(
+      (newTheme, color) => `${newTheme} --color-${color}: ${theme[color as keyof typeof theme]};`,
+      '',
     )
-  }, [colorOverride])
 
-  return (
-    <div className={theme === 'dark' ? 'theme-dark' : 'theme-light'} style={colorOverrideStyles}>
-      {children}
-    </div>
-  )
+    document.body.style.cssText = themeStyles
```

Comment:
What happens if there is more than one ThemeProvider with different values for `theme`?

Reply:
As the ThemeProvider stands right now it only applies styles to the body so if you had more than one they would just override each other.

However I have been toying with a feature that would allow you to supply the theme provider with a ref so the theme provider could apply the theme and dark mode styles to any container. This allows you to have more than one Theme Provider and anything applied to the child theme provider overwrites the body styles. Here is a quick test of this in action:

<img width="711" alt="CleanShot 2022-07-11 at 11 51 59@2x" src="https://user-images.githubusercontent.com/20915535/178337544-aa1cea42-8c63-44f0-9e66-f9d2866c10ab.png">

The top slider is being styled by the standard theme provider and the bottom slider has its own theme provider that is always set to dark mode and has it own theme.